src/lib/stepper/stepper-vertical.html

```diff
@@ -6,19 +6,28 @@
        [tabIndex]="_focusIndex == i ? 0 : -1"
        (click)="step.select()"
        (keydown)="_onKeydown($event)">
-    <div class="mat-stepper-index">
+    <div class="mat-stepper-index-new" *ngIf="!step.completed || selectedIndex == i">
       {{i + 1}}
     </div>
+    <div class="mat-stepper-index-interacted" *ngIf="step.completed && selectedIndex != i">
```

Comment:
@kara Sorry, what do you mean by sharing the icon group? Do you mean like creating a separate icon group component with the template?